dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Access.java

```diff
@@ -1542,26 +1564,39 @@ private boolean isAccessAuthorized(DataFile df, String apiToken) {
         return false;
     }
 
+    private boolean isRestrictedByEmbargo(Dataset ds) {
+        return ds.hasActiveEmbargo() && !permissionService.on(ds).has(Permission.ViewUnpublishedDataset);
+    }
 
-    private User findAPITokenUser(String apiToken) {
-        User apiTokenUser = null;
+    private boolean isRestrictedByEmbargo(Dataset ds, String apiToken) {
+        Optional<User> apiTokenUser = getApiTokenUser(apiToken);
+        if(!apiTokenUser.isPresent()) {
+            return ds.hasActiveEmbargo() && !permissionService.on(ds).has(Permission.ViewUnpublishedDataset);
+        }
+        return apiTokenUser.map(user -> ds.hasActiveEmbargo() &&
+                    !permissionService.requestOn(createDataverseRequest(user), ds).has(Permission.ViewUnpublishedDataset))
+                .orElse(false);
+    }
 
-        if ((apiToken != null) && (apiToken.length() != 64)) {
-            // We'll also try to obtain the user information from the API token, 
-            // if supplied: 
+    private boolean isRestrictedByEmbargo(Long dataVariableId) {
+        DataVariable dataVariable = findDataVariable(dataVariableId);
+        if(dataVariable == null) {
+            throw new BadRequestException("There is no data variable with id: " + dataVariableId);
+        }
+        Dataset dataFileOwner = dataVariable.getDataTable().getDataFile().getOwner();
+        return isRestrictedByEmbargo(dataFileOwner);
+    }
 
+    private Optional<User> getApiTokenUser(String apiToken) {
+        User apiTokenUser = null;
+        if ((apiToken != null) && (apiToken.length() != 64)) {
```

Comment:
So if we would to remove those checks we could make the code 
```
Try.of(this::findUserOrDie)
                .onFailure(throwable -> logger.log(Level.FINE, "some message", throwable))
                .toJavaOptional();
```
Also passing exception.getmessage is giving less information for debugging, it is better to just pass exception to logger.

Reply:
I wanted to be as consistent with current code as possible. Also checking for token length prevent making obviously unnecessary calls to db. But, yet again, in case of highly unlikely event that we make a change resulting in token length change I guess we can omit this check.

Reply:
Yeah it omits db call but it is much more error prone. If we wanted to support this, we would  probably need to add system wide property. Then you would need to check what kind of token version is set and validate it. So there would needed much more work.